src/rookify/yaml.py

```diff
@@ -1,14 +1,17 @@
 # -*- coding: utf-8 -*-
 
+import importlib.resources
+import importlib.resources.abc
 import yamale
 import yaml
-from importlib import resources
 from pathlib import Path
 from typing import Any, Dict
 
 
-_schema_file = Path("rookify", "config.schema.yaml")
-for entry in resources.files("rookify").iterdir():
+_schema_file: Path | importlib.resources.abc.Traversable = Path(
+    "rookify", "config.schema.yaml"
+)
+for entry in importlib.resources.files("rookify").iterdir():
```

Comment:
What do you want to do here? Why du you initialize _schema_file with Path("rookify", "config.schema.yaml") and then iterate over importlib.resources.files("rookify").iterdir() and looking for a file named "config.schema.yaml"?

First ist type Path, second is type Traversable, so mypy is correct to show an error here, if _schema_file is not defined as Path | Traversable, but I don't get, why you're doing this iteration

Reply:
Well, the logic should support both cases either that code is checked out from git as well as if installed as a wheel file. In fact I would like to install the yaml file to a non-source directory in the future but the current implementation has known limitations in this regard.

Long story short: This implementation expects that the current working directory is `src` from git if the `rookify` package is imported without being installed (otherwise main could not be called anyway). As it should be installed as a Python package and the working directory can be any user selected one containing the `config.yaml` rookify will correctly find the filesystem location of the scheme file iterating over the package resources.